presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewCandidateExtractor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.analyzer;
+
+import com.facebook.presto.sql.tree.DefaultTraversalVisitor;
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.facebook.presto.sql.tree.Table;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public class MaterializedViewCandidateExtractor
+        extends DefaultTraversalVisitor<Void, Void>
+{
+    private final Set<QualifiedName> tableNames = new HashSet<>();
+    private final Map<QualifiedName, List<QualifiedName>> baseTableToMaterializedViews;
+
+    public MaterializedViewCandidateExtractor(Map<QualifiedName, List<QualifiedName>> baseTableToMaterializedViews)
+    {
+        this.baseTableToMaterializedViews = requireNonNull(baseTableToMaterializedViews, "baseTableToMaterializedViews is null");
+    }
+
+    @Override
+    protected Void visitTable(Table node, Void context)
+    {
+        tableNames.add(node.getName());
+        return null;
+    }
+
+    public Optional<QualifiedName> getMaterializedViewCandidate()
```

Comment:
Lets use QualifiedObjectName everywhere.

Reply:
The reason i implemented QualifiedName instead of QualifiedObjectName here is that creating QualifiedObjectName requires session. Therefore, I would have to create both queryrunner and session in my test cases, which would be more complicated. Also, this PR merely tests the correctness of candidate extractor, and creating both queryrunner and session in the test case might defeat the purpose of independent testing.

Reply:
You can create QualifiedObjectName using following constructor. 
public static QualifiedObjectName valueOf(String catalogName, String schemaName, String objectName)